### PR TITLE
EDL add safeguards

### PR DESCRIPTION
**Pull request by ilaner**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/46470

## Description
Add safeguards to EDL

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/EDL/Integrations/EDL/EDL.py [hunk 1]
```diff
@@ -109,7 +114,9 @@ def __init__(self,
                  fields_to_present: str = '',
                  csv_text: bool = False,
                  url_protocol_stripping: bool = False,
-                 url_truncate: bool = False
+                 url_truncate: bool = False,
+                 maximum_cidr_size: int = 8,
+                 no_tld: bool = False,
```
Comment (DeanArbel) [inline Packs/EDL/Integrations/EDL/EDL.py:119]: Perhaps rename this to `no_tld_glob`.

File: Packs/EDL/Integrations/EDL/EDL.py [hunk 2]
```diff
@@ -109,7 +114,9 @@ def __init__(self,
                  fields_to_present: str = '',
                  csv_text: bool = False,
                  url_protocol_stripping: bool = False,
-                 url_truncate: bool = False
+                 url_truncate: bool = False,
+                 maximum_cidr_size: int = 8,
```
Comment (DeanArbel) [inline Packs/EDL/Integrations/EDL/EDL.py:118]: Please extract this to a constant.

File: Packs/EDL/Integrations/EDL/EDL.yml [hunk 3]
```diff
@@ -70,6 +70,20 @@ configuration:
   name: add_comment_if_empty
   required: false
   type: 8
+- additionalinfo: 'CIDRs with prefix size smaller than this value will not be included'
```
Comment (ytauman) [inline Packs/EDL/Integrations/EDL/EDL.yml:73]: CIDRs with a smaller network prefix bits number will not be included. For example - if the number is 8, then 0.0.0.0/2 will be excluded from the list.

File: Packs/EDL/Integrations/EDL/EDL.yml [hunk 4]
```diff
@@ -70,6 +70,20 @@ configuration:
...
+  defaultvalue: 8
+  display: 'Maximum CIDR prefix size'
+  hidden: false
+  name: maximum_cidr_size
+  required: true
+  type: 0
+- additionalinfo: 'Configure to disable top level domain in the list.'
```
Comment (DeanArbel) [inline Packs/EDL/Integrations/EDL/EDL.yml:80]:
```suggestion
- additionalinfo: 'Configure to drop top level domain globs that would've been added to the list.'
```

File: Packs/EDL/Integrations/EDL/EDL.yml [hunk 5]
```diff
@@ -70,6 +70,20 @@ configuration:
   name: add_comment_if_empty
   required: false
   type: 8
+- additionalinfo: 'CIDRs with prefix size smaller than this value will not be included'
+  defaultvalue: 8
+  display: 'Maximum CIDR prefix size'
```
Comment (ytauman) [inline Packs/EDL/Integrations/EDL/EDL.yml:75]: CIDR network prefix bits size

File: Packs/EDL/Integrations/EDL/EDL.yml [hunk 6]
```diff
@@ -70,6 +70,20 @@ configuration:
...
+  hidden: false
+  name: maximum_cidr_size
+  required: true
+  type: 0
+- additionalinfo: 'Configure to disable top level domain in the list.'
+  defaultvalue: 'false'
+  display: 'Disable insertion top level domains'
```
Comment (DeanArbel) [inline Packs/EDL/Integrations/EDL/EDL.yml:82]: This display is a bit too technical. The user doesn't think of EDL as if it inserts results into a list. Perhaps `Drop top level domain globs`?

File: Packs/EDL/Integrations/EDL/EDL.yml (hunk 4 above)
Comment (ytauman) [inline Packs/EDL/Integrations/EDL/EDL.yml:80]: Option to remove top level domainGlobs from the list. For example - *.com.

File: Packs/EDL/Integrations/EDL/EDL.yml (hunk 6 above)
Comment (ytauman) [inline Packs/EDL/Integrations/EDL/EDL.yml:82]: Exclude top level domainGlobs

File: Packs/EDL/Integrations/EDL/EDL_test.py [hunk 7]
```diff
@@ -195,8 +196,43 @@ def test_create_new_edl(self, mocker):
                         'iopqwertyuiopqwertyuiopqwertyuiopqwertyuiopqwertyuiopqwertyuiopqwertyuiopqwertyuiopyuioplkjhgfdsa' \
                         'zxqwertyuiopqwertyuiopqwertyuiopqwertyuiopqwertyuiopqwertyuiopqwertyuiopqwertyuiopqwertyuiopqwert' \
                         'yuiopqwertyuiopqwertyuiopqwertyuiop", "indicator_type": "URL"}\n' \
-                        '{"value": "demisto.com", "indicator_type": "URL"}' \
+                        '{"value": "demisto.com", "indicator_type": "URL"}'
 
+    def test_create_new_edl_edge_cases(self, mocker, requests_mock):
```
Comment (DeanArbel) [inline Packs/EDL/Integrations/EDL/EDL_test.py:201]: Please add a docstring in the style of `Summary, Given, When Then`.
It's best to document what are the **edge cases** that you are testing.

File: Packs/EDL/Integrations/EDL/README.md [hunk 8]
```diff
@@ -27,36 +27,44 @@ Unlike `PAN-OS EDL Management`, this integration hosts the EDL on the Cortex XSO
...
+| Show CSV formats as Text           | If selected, CSV format will appear in a textual webpage instead of initiating a file download.                                                                                                                                                      | False        |
+| XSOAR Indicator Page Size          | Internal page size used when querying Cortex XSOAR for the indicators.                                                                                                                                                                               | False        |
+| Maximum CIDR prefix size   | CIDRs with prefix size smaller than this value will not be included. Default is 8.                                                                                                                                                                   | True         |
```
Comment (ytauman) [inline Packs/EDL/Integrations/EDL/README.md:56]: CIDR network prefix bits size | CIDRs with a smaller network prefix bits number will not be included. For example - if the number is 8, then 0.0.0.0/2 will be excluded from the list. | False

File: Packs/EDL/Integrations/EDL/README.md [hunk 9]
```diff
@@ -27,36 +27,44 @@ Unlike `PAN-OS EDL Management`, this integration hosts the EDL on the Cortex XSO
...
+| Disable insertion top level domains        | Check if disable top Level Domain in the list. Default is False.                                                                                                                                                                                     | True         |
```
Comment (ytauman) [inline Packs/EDL/Integrations/EDL/README.md:57]: Exclude top level domainGlobs | Option to remove top level domainGlobs from the list. For example - *.com. | False

File: Packs/EDL/Integrations/EDL/README.md [hunk 10]
```diff
@@ -27,36 +27,44 @@ Unlike `PAN-OS EDL Management`, this integration hosts the EDL on the Cortex XSO
...
+| Advanced: NGINX Server Conf        | NGINX server configuration to be used instead of the default NGINX_SERVER_CONF used in the integration code. Advanced configuration to be used only if instructed by XSOAR Support.                                                                  | False        |
+| Advanced: NGINX Read Timeout       | NGNIX read timeout in seconds.                                                                                                                                                                                                                       | False        |
+| Advanced: use legacy queries       | When enabled, the integration will query the server using full queries. Advanced configuration to be used only if instructed by XSOAR Support, or you've encountered log errors in the form of: 'msgpack: invalid code.'                             | False        |
+
+### Safeguards
```
Comment (ytauman) [inline Packs/EDL/Integrations/EDL/README.md:63]: There are two integrations parameters that are used as safeguards: `CIDR network prefix bits size` and `Exclude top level domainGlobs`.
Those are meant to prevent the integration to incorrectly insert unwanted TLDs or too wide of a range in a CIDR.
The default value for `CIDR network prefix bits size` is 8, which means that CIDRs with a smaller network prefix bits number will not be included (such as `0.0.0.0/2`).
The default value for `Exclude top level domainGlobs` is off. If enabled, the exported list will not hold indicators such as `*.com`, `*.co.uk`, `*.org` and other top level domains.

File: Packs/EDL/Integrations/EDL/README.md [hunk 11]
```diff
@@ -27,36 +27,44 @@ Unlike `PAN-OS EDL Management`, this integration hosts the EDL on the Cortex XSO
...
+There are two integrations parameters that are used as safeguards: `Maximum CIDR prefix size` and `Disable insertion top level domains`.
+Those are meant to prevent the integration to incorrectly insert Top Level Domains or too wide CIDRs.
+The default value for `Maximum CIDR prefix size` is 8, which means that CIDR's of prefix smaller than 8 will not be inserted. This option is configurable by the user.
+The default value for `Disable insertion top level domains` is off. If enabled, EDL will not insert items such as `*.com`, `*.co.uk`, `*.org` and all the Top Level Domains.
```
Comment (DeanArbel) [inline Packs/EDL/Integrations/EDL/README.md:67]:
```suggestion
The default value for `Drop top level domains globs` is off. If enabled, EDL will drop items such as `*.com`, `*.co.uk`, `*.org` and all other Top Level Domains.
```

File: Packs/EDL/Integrations/EDL/EDL.yml [hunk 12]
```diff
@@ -70,6 +70,20 @@ configuration:
   name: add_comment_if_empty
   required: false
   type: 8
+- additionalinfo: 'CIDRs with prefix size smaller than this value will not be included'
+  defaultvalue: 8
+  display: 'Maximum CIDR prefix size'
+  hidden: false
+  name: maximum_cidr_size
+  required: true
```
Comment (ytauman) [inline Packs/EDL/Integrations/EDL/EDL.yml:78]: why is this required?

File: Packs/EDL/Integrations/EDL/EDL.yml [hunk 13]
```diff
@@ -70,6 +70,20 @@ configuration:
...
+  type: 0
+- additionalinfo: 'Configure to disable top level domain in the list.'
+  defaultvalue: 'false'
+  display: 'Disable insertion top level domains'
+  hidden: false
+  name: no_tld
+  required: true
```
Comment (ytauman) [inline Packs/EDL/Integrations/EDL/EDL.yml:85]: why is this required?

File: Packs/EDL/Integrations/EDL/EDL.yml (hunk 3 above)
Comment (ytauman) [reply Packs/EDL/Integrations/EDL/EDL.yml:73]: i would like to move this option definition so that it's right after 'IP Collapsing' option

File: Packs/EDL/Integrations/EDL/EDL.yml (hunk 4 above)
Comment (ytauman) [reply Packs/EDL/Integrations/EDL/EDL.yml:80]: same here - after 'CIDR network prefix bits size', after 'IP Collapsing'

File: Packs/EDL/ReleaseNotes/3_1_0.md [hunk 14]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### Generic Export Indicators Service
```
Comment (ytauman) [inline Packs/EDL/ReleaseNotes/3_1_0.md:3]: needs to be updated here too

**Review by ytauman — COMMENTED** [review]
some string changes, make sure to update them everywhere they are referenced.
also - not sure the params need to be mandatory.

**Review by DeanArbel — APPROVED** [review]
Very good and clean job. I have several comments, but they're mostly stylings.
@jrosenpanw Please assign a technical writer to go over this with @ilaner .

**Comment by bakatzir** [comment]
was there a demo @ilaner ? Dror and myself would love to see it briefly.

**Comment by Ni-Knight** [comment]
@bakatzir I was in the demo.

File: Packs/EDL/Integrations/EDL/EDL.py (hunk 1 above)
Comment (Ni-Knight) [reply Packs/EDL/Integrations/EDL/EDL.py:119]: `no_wildcard_tld` might be a bit better (as wildcard is the common term for it)

File: Packs/EDL/Integrations/EDL/EDL.yml (hunk 5 above)
Comment (Ni-Knight) [reply Packs/EDL/Integrations/EDL/EDL.yml:75]: I actually like `Maximum CIDR prefix size` a bit better, we can join the two to `Maximum CIDR network prefix bits size`

**Comment by richardbluestone** [comment]
> Very good and clean job. I have several comments, but they're mostly stylings. @jrosenpanw Please assign a technical writer to go over this with @ilaner .

@ilaner Are there specific strings you want me to review, or do you want me to review generally?

**Comment by tschanfeld** [comment]
edited yml and md files
